src/main/java/appeng/core/sync/packets/PacketNEIRecipe.java

```diff
@@ -62,117 +61,297 @@
 import appeng.util.prioitylist.IPartitionList;
 
 
-public class PacketNEIRecipe extends AppEngPacket
+public class PacketNEIRecipe implements AppEngPacket, AppEngPacketHandler<PacketNEIRecipe, AppEngPacket>
 {
 
-	ItemStack[][] recipe;
+	private ItemStack[][] recipeStack;
+	private NBTTagCompound recipe;
 
-	// automatic.
-	public PacketNEIRecipe( ByteBuf stream ) throws IOException
+	@Reflected
+	public PacketNEIRecipe()
 	{
-		ByteArrayInputStream bytes = new ByteArrayInputStream( stream.array() );
-		bytes.skip( stream.readerIndex() );
-		NBTTagCompound comp = CompressedStreamTools.readCompressed( bytes );
-		if( comp != null )
+		// automatic.
+	}
+
+	// api
+	public PacketNEIRecipe( NBTTagCompound recipe ) throws IOException
+	{
+		this.recipe = recipe;
+	}
+
+	// @Override
+	// public void serverPacketData( INetworkInfo manager, AppEngPacket packet, EntityPlayer player )
+	// {
+	// EntityPlayerMP pmp = (EntityPlayerMP) player;
+	// Container con = pmp.openContainer;
+	//
+	// if( con instanceof IContainerCraftingPacket )
+	// {
+	// IContainerCraftingPacket cct = (IContainerCraftingPacket) con;
+	// IGridNode node = cct.getNetworkNode();
+	// if( node != null )
+	// {
+	// IGrid grid = node.getGrid();
+	// if( grid == null )
+	// {
+	// return;
+	// }
+	//
+	// IStorageGrid inv = grid.getCache( IStorageGrid.class );
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACommentedOutCodeLine)